openmetadata-ui/src/main/resources/ui/src/components/GlossaryTerms/tabs/GlossaryTermSynonyms.tsx

```diff
@@ -59,7 +59,8 @@ const GlossaryTermSynonyms = ({
 
   useEffect(() => {
     if (glossaryTerm.synonyms?.length) {
-      setSynonyms(glossaryTerm.synonyms);
+      // removing empty string
+      setSynonyms(glossaryTerm.synonyms.filter((synonym) => !isEmpty(synonym)));
```

Comment:
We are getting the glossary term via props and assigning the initial value of synonyms in useEffect!
Alos we need to maintain state as it will also update with onChange function!